scripts/jenkins/build-test.sh

```diff
@@ -5,4 +5,16 @@ set -euxo pipefail
 echo "Installing ${GO_VERSION} with gimme."
 eval "$(curl -sL https://raw.githubusercontent.com/travis-ci/gimme/master/gimme | GIMME_GO_VERSION=${GO_VERSION} bash)"
 
-make install check
+make install precheck check-modules
+
+# Run the tests
+set +e
+export OUT_FILE="build/test-report.out"
+mkdir -p build
+make test 2>&1 | tee ${OUT_FILE}
+status=$?
+
+go get -v -u github.com/jstemmer/go-junit-report
+go-junit-report > "build/junit-apm-agent-go-${GO_VERSION}.xml" < ${OUT_FILE}
```

Comment:
Isn't this similar to what we do here: https://github.com/elastic/apm-agent-go/blob/master/scripts/docker-test.sh#L19?

Reply:
Indeed

Reply:
In other words, the current JUnit reporting is only enabled for the coverage that for some reason uses `docker-compose`. While the test stage uses the CI Workers context instead a docker container.

For such, the implementation details are different, that particular `docker-test.sh`, it was created to align the same steps for travis and jenkins: 
- https://github.com/elastic/apm-agent-go/pull/588

Reply:
So at some point, when we run this only on Jenkins, we'll align scripts I guess